src/services/cronJobs/fillSnapshotBalances.ts

```diff
@@ -46,88 +49,122 @@ export async function addFillPowerSnapshotBalanceJobsToQueue() {
       });
+      isFinished = true;
+      break;
     }
-  });
+    offset += powerBoostings.length;
+
+    // logger.debug('**powerBoostings**', powerBoostings);
+    logger.debug('groupByTimestamp', groupByTimestamp);
+    Object.keys(groupByTimestamp).forEach(key => {
+      const powerSnapshotId = groupByTimestamp[key][0].powerSnapshotId;
+      logger.debug('inside forEach', {
+        timestamp: Number(key),
+        powerSnapshotId,
+      });
+      const jobData = {
+        timestamp: Number(key),
+        // timestamp: Number(Math.floor(Number(key) / 1000)),
+        powerSnapshotId,
+        data: groupByTimestamp[key].map(pb => {
+          return {
+            userId: pb.userId,
+            walletAddress: pb.walletAddress.toLowerCase(),
+          };
+        }),
+      };
+      logger.debug('job data', jobData);
+      fillSnapshotBalanceQueue.add(jobData);
+    });
+  }
 }
 
 export function processFillPowerSnapshotJobs() {
   logger.debug('processFillPowerSnapshotJobs() has been called ', {
-    numberOfFillPowerSnapshotBAlancesConcurrentJob,
+    numberOfFillPowerSnapshotBAlancesConcurrentJob:
+      numberOfFillPowerSnapshotBalancesConcurrentJob,
   });
   fillSnapshotBalanceQueue.process(
-    numberOfFillPowerSnapshotBAlancesConcurrentJob,
+    numberOfFillPowerSnapshotBalancesConcurrentJob,
     async (job, done) => {
-      logger.debug('processing fill powerSnapshot job', job.data);
-      const { blockNumber, data } = job.data;
+      logger.debug('*************hi***');
+      const items = job.data.data;
+      const { timestamp, powerSnapshotId } = job.data;
       try {
+        const addresses = items.map(item => item.walletAddress).join(',');
+        // logger.debug('processFillPowerSnapshotJobs() addresses ', {
+        //   addresses,
+        //   timestamp,
+        //   powerSnapshotId,
+        // });
         const balances =
-          await getGivPowerSubgraphAdapter().getUserPowerBalanceAtBlockNumber({
-            blockNumber,
-            walletAddresses: data.map(user =>
-              user?.walletAddress?.toLowerCase(),
-            ),
+          await getPowerBalanceAggregatorAdapter().getBalanceOfAddresses({
+            timestamp,
+            addresses,
           });
 
+        // logger.debug('addresses and balances', {
+        //   addresses,
+        //   balances,
+        // });
+
         await addOrUpdatePowerSnapshotBalances(
-          data.map(item => {
+          balances.map(balance => {
             return {
-              balance: balances[item.walletAddress.toLowerCase()].balance,
-              powerSnapshotId: item.powerSnapshotId,
-              userId: item.userId,
+              balance: balance.balance,
+              powerSnapshotId,
+              userId: items.find(
+                item =>
+                  item.walletAddress.toLowerCase() ===
+                  balance.address.toLowerCase(),
+              )!.userId,
             };
           }),
         );
```

Comment:
What if the balance of some users are not returned!

Reply:
It's not important we find through our addresses from DB not balances , and we are sure we find a user because we just get balances of addresses that we have sent

Reply:
It must have the balance! otherwise there would be an issue... let's keep it as it is